acquire/guis/gui.py

```diff
@@ -0,0 +1,100 @@
+from __future__ import annotations
+
+import logging
+import time
+from argparse import Namespace
+from threading import Thread
+
+log = logging.getLogger("gui")
+
+
+class GUI:
+    _instance = None
+    _progress = 0
+    _closed = False
+    _shard = 0
+
+    thread = None
+    folder = None
+    ready = False
+    auto_upload = None
+    upload_available = False
+
+    def __new__(cls, flavour: str = None, upload_available: bool = False):
+        # singleton+factory pattern
+        if cls._instance is None:
+            if flavour == "Windows":
+                # create a basic Win32 GUI
+                from acquire.guis.wingui import WinGUI
+
+                cls = WinGUI
+                log.info("Creating win32 gui instance")
+            else:
+                # Use the NULL-pattern here, to avoid many IFs
+                cls = StubGUI
+                log.info("Creating stub gui instance")
+
+            if flavour is None:
+                log.warning("GUI has been initialised with invalid flavour, possible logic flaw.")
+
```

Comment:
Shouldn't we create list (or enum) of possible `flavours`? I understand its only Windows. But this will likely change in the future. 
```suggestion
            if flavour not in flavours:
                log.warning("GUI has been initialised with invalid flavour (%s).", flavour)
```